packages/react/plugins/component-testing/index.ts

```diff
@@ -81,29 +93,46 @@ export function nxComponentTestingPreset(
       );
     }
 
-    webpackConfig = buildTargetWebpack(graph, buildTarget, ctProjectName);
+    if (!options.viteConfig) {
+      webpackConfig = buildTargetWebpack(graph, buildTarget, ctProjectName);
+    }
   } catch (e) {
     logger.warn(
       stripIndents`Unable to build a webpack config with the project graph. 
       Falling back to default webpack config.`
     );
     logger.warn(e);
-    webpackConfig = buildBaseWebpackConfig({
-      tsConfigPath: 'cypress/tsconfig.cy.json',
-      compiler: 'babel',
-    });
+    if (!options.viteConfig) {
+      webpackConfig = buildBaseWebpackConfig({
+        tsConfigPath: 'cypress/tsconfig.cy.json',
+        compiler: 'babel',
+      });
+    }
+  }
+
+  if (options.viteConfig) {
+    return {
+      ...nxBaseCypressPreset(pathToConfig),
+      specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
+      devServer: {
+        framework: 'react',
+        bundler: 'vite',
+        viteConfig: options.viteConfig,
+      } as const,
+    };
+  } else {
+    return {
+      ...nxBaseCypressPreset(pathToConfig),
```

Comment:
this can all be simplified into something like 
```
    if (options.viteConfig) {
      return {
        ...nxBaseCypressPreset(pathToConfig),
        specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
        devServer: {
          ...({ framework: 'react', bundler: 'vite' } as const),
        },
      };
    }

// TODO: lazy load the webpack specific imports needed so vite users don't need @nrwl/webpack installed
    webpackConfig = buildTargetWebpack(graph, buildTarget, ctProjectName);
  } catch (e) {
    logger.warn(
      stripIndents`Unable to build a webpack config with the project graph. 
      Falling back to default webpack config.`
    );
    logger.warn(e);
    webpackConfig = buildBaseWebpackConfig({
      tsConfigPath: 'cypress/tsconfig.cy.json',
      compiler: 'babel',
    });
  }

  return {
    ...nxBaseCypressPreset(pathToConfig),
    specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
    devServer: {
      // cypress uses string union type,
      // need to use const to prevent typing to string
      ...({
        framework: 'react',
        bundler: 'webpack',
      } as const),
      webpackConfig,
    },
  };
```

Reply:
and instead of options.viteConfig we can look at the buildTarget to see what executor is being used/if there is a vite.config.{js,ts} in the projectRoot and default to vite if so then fall back to webpack. etc. thought maybe too much magic 🤔

Reply:
Automatically detect!